swim/handlers_test.go

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package swim
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/stretchr/testify/suite"
+	"github.com/uber/ringpop-go/shared"
+	"github.com/uber/ringpop-go/swim/test/mocks"
+	"github.com/uber/tchannel-go"
+)
+
+type HandlerTestSuite struct {
+	suite.Suite
+
+	testNode *testNode
+	peers    []*testNode
+
+	cluster *swimCluster
+
+	ctx       tchannel.ContextWithHeaders
+	ctxCancel context.CancelFunc
+}
+
+func (s *HandlerTestSuite) SetupTest() {
+	// Create a test node, on its own. Tests can join this to the cluster for
+	// testing, if they want.
+	s.testNode = newChannelNode(s.T())
+	s.peers = append(s.peers, s.testNode)
+
+	// Create a cluster for testing. Join these guys to each other.
+	s.cluster = newSwimCluster(4)
+	s.cluster.Bootstrap()
+
+	s.ctx, s.ctxCancel = shared.NewTChannelContext(500 * time.Millisecond)
+}
+
+func (s *HandlerTestSuite) TearDownTest() {
+	s.ctxCancel()
+
+	// Destroy any nodes added to the peer list during testing
+	for _, node := range s.peers {
+		if node != nil {
+			node.Destroy()
+		}
+	}
+	if s.cluster != nil {
+		s.cluster.Destroy()
+	}
+}
+
+func (s *HandlerTestSuite) TestGossipStartHandler() {
+	s.testNode.node.gossip.Stop()
```

Comment:
This is another example of handler tests that I do not necessarily agree with. I don't want you to change it if it will add more time to completion, but I want to make note of it. I feel the test being performed here should be defined where the `gossip` object is the subject under test (SUT), aka `swim/gossip_test.go`. 

Nevertheless, it's good to have coverage here. And I appreciate that you've done the work. :+1: